addons/xterm-addon-serialize2/src/serializer.wasm.ts

```diff
@@ -0,0 +1,214 @@
+    #define RED_SHIFT         ${Attributes.RED_SHIFT}
+
+    /* bit 25..26   color mode: DEFAULT (0) | P16 (1) | P256 (2) | RGB (3) */
+    #define CM_MASK           ${Attributes.CM_MASK}
+    #define CM_DEFAULT        ${Attributes.CM_DEFAULT}
+    #define CM_P16            ${Attributes.CM_P16}
+    #define CM_P256           ${Attributes.CM_P256}
+    #define CM_RGB            ${Attributes.CM_RGB}
+
+    /* bit 1..24  RGB room */
+    #define RGB_MASK          ${Attributes.RGB_MASK}
+    #define COLOR_MASK        ${Attributes.CM_MASK | Attributes.RGB_MASK}
+
+    /* fg flags:   bit 27..32 */
+    #define INVERSE           ${FgFlags.INVERSE}
+    #define BOLD              ${FgFlags.BOLD}
+    #define UNDERLINE         ${FgFlags.UNDERLINE}
+    #define BLINK             ${FgFlags.BLINK}
+    #define INVISIBLE         ${FgFlags.INVISIBLE}
+    #define STRIKETHROUGH     ${FgFlags.STRIKETHROUGH}
+
+    /* bg flags:   bit 27..32 (upper 2 unused) */
+    #define ITALIC            ${BgFlags.ITALIC}
+    #define DIM               ${BgFlags.DIM}
+    #define HAS_EXTENDED      ${BgFlags.HAS_EXTENDED}
+    #define PROTECTED         ${BgFlags.PROTECTED}
+
+    /* ext flags:   bit 27..32 (upper 3 unused) */
+    #define UNDERLINE_STYLE   ${ExtFlags.UNDERLINE_STYLE}
+
+    /* underline style */
+    #define UL_NONE           ${UnderlineStyle.NONE}
+    #define UL_SINGLE         ${UnderlineStyle.SINGLE}
+    #define UL_DOUBLE         ${UnderlineStyle.DOUBLE}
+    #define UL_CURLY          ${UnderlineStyle.CURLY}
+    #define UL_DOTTED         ${UnderlineStyle.DOTTED}
+    #define UL_DASHED         ${UnderlineStyle.DASHED}
+
+
+    /* memory locations */
+    #define P_LUT100 256
+    #define P_EXT 16384
+
+    ${require('fs').readFileSync('src-wasm/serialize.c')}
+  `
+});
+
+// itoa LUT
+// note: performant decimal conversion of numbers is actually a hard problem
+// we use a LUT approach with 0-99 precomputed in utf16 for better performance
+const LUT100 = new Uint32Array(100);
+for (let i1 = 0; i1 < 10; ++i1) {
+  for (let i2 = 0; i2 < 10; ++i2) {
+    LUT100[i1 * 10 + i2] = (48 + i2) << 16 | (48 + i1);
+  }
+}
+
+const mem = new WebAssembly.Memory({ initial: 3000 });
+const d16 = new Uint16Array(mem.buffer);
+const d32 = new Uint32Array(mem.buffer);
+let writeCombinedOverload: (dst: number, x: number) => number = (dst, x) => dst;
+const writeCombined: (dst: number, x: number) => number = (dst, x) => writeCombinedOverload(dst, x);
+let writeLinkOverload: (dst: number, x: number) => number = (dst, x) => dst;
+const writeLink: (dst: number, x: number) => number = (dst, x) => writeLinkOverload(dst, x);
+const inst = wasmSerialize({ env: { memory: mem, writeCombined, writeLink } });
+d32.set(LUT100, 64);
+const td = new TextDecoder('UTF-16LE');
+
+
+export function serialize(t: ITerminal): string {
+  // reset FG/BG/ext
+  inst.exports.reset(0, 0, 0, 0);
+
+  const buffer = t.buffer;
+  const len = t.buffer.lines.length;
+  let wPos = 150*65536;
```

Comment:
Surprised this isn't a lint rule:

```suggestion
  let wPos = 150 * 65536;
```

Reply:
Hmm, interesting. Yeah I fixed all linter nagging, but this one did not show up.

The hardcoded values will be replaced by proper memory management, prolly static again to avoid wasm compilation over and over, instead calling earlier into `TextDecoder.decode`.